src/main/java/duke/Parser.java

```
@@ -0,0 +1,209 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+/**
+ * Represents commands from user input
+ */
+public class Parser {
+    private static final String Indentation = " ";
+    private static final String Horizontal = "____________________________________________________________";
+    private Ui ui;
+
+    /**
+     * Parser constructor
+     * @param ui
+     */
+    public Parser(Ui ui) {
+        this.ui = ui;
+    }
+
+    /**
+     * method for list, mark, unmark, delete, to do, deadline, event commands
+     * @param cmd
+     * @param  tasks
+     * @return command
+     */
+    public static String parse(String cmd, TaskList tasks) {
+        String command = cmd.trim();
+        String[] words = command.split(" ");
+        Task task = null;
+        String info;
+        String str = "";
+
+        if (!command.equals("bye")) {
+            if (command.equals("list")) {
+                //Ui.showList(tasks);
+                str += Ui.showList(tasks);
+
+            } else if (words[0].equals("mark")) {
+                try {
+                   //Ui.done(words[1], tasks);
+                   str += Ui.done(words[1], tasks);
+
+                } catch (Exception e) {
+                    System.out.println(Indentation + Horizontal);
+                    System.out.println("  ☹ OOPS!!! The index number cannot be empty.");
+                    System.out.println(Indentation + Horizontal);
+
+                    str += "  ☹ OOPS!!! The index number cannot be empty. \n";
```

To avoid having to repeat your error messages, you can possibly throw say an `InvalidInputException` with your intended error message, and then handle it in the Duke class. The section on [A-MoreOOP](https://nus-cs2103-ay2223s2.github.io/website/se-book-adapted/projectDuke/index.html#a-moreoop) provides an example for my explanation.